modules/launcher/src/main/java/org/mule/runtime/module/launcher/application/ApplicationExtensionsManagerConfigurationBuilder.java

```diff
@@ -6,88 +6,93 @@
- * that register a {@link ExtensionManager} if it's present in the classpath
+ * Implementation of {@link ConfigurationBuilder} that registers a {@link ExtensionManager}
  *
  * @since 4.0
  */
 public class ApplicationExtensionsManagerConfigurationBuilder extends AbstractConfigurationBuilder
 {
 
-    private static Log logger = LogFactory.getLog(ApplicationExtensionsManagerConfigurationBuilder.class);
+    private static Logger LOGGER = LoggerFactory.getLogger(ApplicationExtensionsManagerConfigurationBuilder.class);
 
+    private final ExtensionManagerFactory extensionManagerFactory;
     private final List<ApplicationPlugin> applicationPlugins;
 
     public ApplicationExtensionsManagerConfigurationBuilder(List<ApplicationPlugin> applicationPlugins)
     {
+        this(applicationPlugins, new DefaultExtensionManagerFactory());
+    }
+
+    public ApplicationExtensionsManagerConfigurationBuilder(List<ApplicationPlugin> applicationPlugins, ExtensionManagerFactory extensionManagerFactory)
+    {
         this.applicationPlugins = applicationPlugins;
+        this.extensionManagerFactory = extensionManagerFactory;
     }
 
     @Override
     protected void doConfigure(MuleContext muleContext) throws Exception
     {
-        ExtensionManager extensionManager = new DefaultExtensionManager();
-        ((DefaultMuleContext) muleContext).setExtensionManager(extensionManager);
-        initialiseIfNeeded(extensionManager, muleContext);
+        final ExtensionManagerAdapter extensionManager = createExtensionManager(muleContext);
 
         for (ApplicationPlugin applicationPlugin : applicationPlugins)
         {
-            final ServiceLoader<Describer> describers = ServiceLoader.load(Describer.class, new ClassLoader(null)
+            URL manifestUrl = applicationPlugin.getArtifactClassLoader().findResource("META-INF/" + EXTENSION_MANIFEST_FILE_NAME);
+            if (manifestUrl == null)
             {
-                @Override
-                public Class<?> loadClass(String name) throws ClassNotFoundException
-                {
-                    return applicationPlugin.getArtifactClassLoader().getClassLoader().loadClass(name);
-                }
+                continue;
+            }
 
-                @Override
-                protected URL findResource(String name)
-                {
-                    return applicationPlugin.getArtifactClassLoader().findResource(name);
-                }
+            ExtensionManifest extensionManifest = extensionManager.parseExtensionManifestXml(manifestUrl);
+            extensionManager.registerExtension(extensionManifest, applicationPlugin.getArtifactClassLoader().getClassLoader());
 
-                @Override
-                protected Enumeration<URL> findResources(String name) throws IOException
-                {
-                    return applicationPlugin.getArtifactClassLoader().findResources(name);
-                }
-            });
+            if (LOGGER.isDebugEnabled())
```

Comment:
This logger should be right after you detected that the plugin contains an extension to make easier to detect problems when the extension parsing fails